client/engine/chainservice/eth_chainservice.go

```diff
@@ -0,0 +1,110 @@
+package chainservice
+
+import (
+	"context"
+	"log"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	ethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	NitroAdjudicator "github.com/statechannels/go-nitro/client/engine/chainservice/adjudicator"
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/types"
+)
+
+type eventSource interface {
+	SubscribeFilterLogs(ctx context.Context, query ethereum.FilterQuery, ch chan<- ethTypes.Log) (ethereum.Subscription, error)
+}
+
+type EthChainService struct {
+	in  chan protocols.ChainTransaction
+	out chan Event
+}
+
+// NewEthChainService constructs a chain service that submits transactions to a NitroAdjudicator
+// and listens to events from an eventSource
+func NewEthChainService(na *NitroAdjudicator.NitroAdjudicator, naAddress common.Address, to *bind.TransactOpts, es eventSource) EthChainService {
+	ecs := EthChainService{}
+	ecs.in = make(chan protocols.ChainTransaction)
+	ecs.out = make(chan Event)
+
+	go ecs.submitTxs(na, to)
+	go ecs.listenForLogEvents(na, naAddress, es)
+
+	return ecs
+}
+
+func (cc EthChainService) In() chan<- protocols.ChainTransaction {
+	return cc.in
+}
+
+func (cc EthChainService) Out(a types.Address) <-chan Event {
+	return cc.out
+}
+
+func (cc EthChainService) submitTxs(na *NitroAdjudicator.NitroAdjudicator, to *bind.TransactOpts) {
+	for tx := range cc.in {
+		cc.submitTx(na, to, tx)
+	}
+}
+
+// submitTx responds to the given tx.
+func (cc EthChainService) submitTx(na *NitroAdjudicator.NitroAdjudicator, to *bind.TransactOpts, tx protocols.ChainTransaction) {
+	switch tx.Type {
+	case protocols.DepositTransactionType:
+		for address, amount := range tx.Deposit {
+			to.Value = amount
+			// TODO do not assume that the channel holds 0 funds
```

Comment:
Let's make an issue to track this?

Reply:
Good call! https://github.com/statechannels/go-nitro/issues/712